### PR TITLE
Converts avatar component to hooks

### PR DESCRIPTION
**Pull request by Chrischuck**
Pretty straight forward port from `PureComponent` to `hooks`

Just make sure it works in storybook + docs. 

**Before: ~19.1kb**
before

**After: ~16.4kb**
Screen Shot 2020-01-28 at 22 46 29

### CODE REVIEW

File: src/avatar/src/Avatar.js [hunk 1]
```diff
@@ -21,167 +21,145 @@ const initialsStyleClass = css({
...
+  color: 'automatic',
+  size: 24,
+  isSolid: false,
+  getInitials: globalGetInitials,
+  forceShowInitials: false,
+  sizeLimitOneCharacter: 20
 }
 
-export default withTheme(Avatar)
+export default Avatar
```
Comment (mshwery) [inline src/avatar/src/Avatar.js]: Same comment as in https://github.com/segmentio/evergreen/pull/724#discussion_r372533724 – for parity we should probably support `forwardRef` + `React.memo`?

File: src/avatar/src/Avatar.js [hunk 2]
```diff
@@ -21,167 +21,145 @@ const initialsStyleClass = css({
...
-    forceShowInitials: false,
-    sizeLimitOneCharacter: 20
-  }
+  return theme.getAvatarProps({ isSolid, color })
+}
 
-  constructor(props, context) {
-    super(props, context)
-    this.state = { imageHasFailedLoading: false }
+function Avatar({
```
Comment (mshwery) [inline src/avatar/src/Avatar.js]: For `defaultProps` should those just be assigned in the method signature? e.g. `forceShowInitials = false`

File: src/avatar/src/Avatar.js [hunk 3]
```diff
@@ -21,167 +21,137 @@ const initialsStyleClass = css({
...
+  forceShowInitials: PropTypes.bool,
+
+  /**
+   * When the size is smaller than this number, use a single initial for the avatar.
+   */
+  sizeLimitOneCharacter: PropTypes.number
 }
 
-export default withTheme(Avatar)
+export default memo(forwardRef(Avatar))
```
Comment (mshwery) [inline src/avatar/src/Avatar.js]: Might be wrong but I think we'll also want to set the `.displayName` of the component (or the forwardRef component... or the memoized component)?

**Comment by Chrischuck** [comment]
> https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers

Seems like a good idea to throw it into v5, curiously, how would rolling out other components work. Since the non hook components would remain the same in v5 but once those components get converted that would be a breaking change in a minor release if that makes sense.

**Comment by mshwery** [comment]
>Since the non hook components would remain the same in v5 but once those components get converted that would be a breaking change in a minor release if that makes sense.

Yeah I think there are a couple ways we could approach this. 
1. Strict mode: We could target these PRs to a `v5` branch (probably need to start fresh since the current one is quite stale) and once we convert _everything_ over... then we finally release v5 to npm
2. Incremental mode: Ship these PRs as v5 release candidates so we can start opting into them without broadly releasing v5. This will allow us to continue to make breaking changes on v5 until we've finished all the incremental PRs. (Though maybe we still want to target a new v5 branch for that instead of `master`?)

**Comment by Chrischuck** [comment]
> > Since the non hook components would remain the same in v5 but once those components get converted that would be a breaking change in a minor release if that makes sense.
> 
> Yeah I think there are a couple ways we could approach this.
> 
> 1. Strict mode: We could target these PRs to a `v5` branch (probably need to start fresh since the current one is quite stale) and once we convert _everything_ over... then we finally release v5 to npm
> 2. Incremental mode: Ship these PRs as v5 release candidates so we can start opting into them without broadly releasing v5. This will allow us to continue to make breaking changes on v5 until we've finished all the incremental PRs. (Though maybe we still want to target a new v5 branch for that instead of `master`?)

Personally, I think incremental mode is the way to go so there's at least some level of real world testing. What're your thoughts?

**Comment by Chrischuck** [comment]
@mshwery this one good to go?

File: src/avatar/src/Avatar.js [hunk 4]
```diff
@@ -156,7 +77,8 @@ class Avatar extends PureComponent {
         justifyContent="center"
         backgroundColor={colorProps.backgroundColor}
         title={name}
-        {...props}
+        innerRef={ref}
```
Comment (mshwery) [inline src/avatar/src/Avatar.js:80]: We'll need to update this to `ref` once https://github.com/segmentio/evergreen/pull/843 lands :)

**Comment by mshwery** [comment]
Thanks for this @Chrischuck ! 🙌

**Pull request merged**